src/app/(main)/(secondary)/search/page.tsx

```diff
@@ -0,0 +1,107 @@
+import { SearchParams, SearchResults, SearchResultsType, SongVersions } from '@/types';
+import { API_DOMAIN } from '@/lib/constants';
+import { SimplePopover } from '@/components/Popover';
+import SearchBar from '@/components/search/SearchBar';
+import SearchFilterPill from '@/components/search/SearchFilterPill';
+import SearchResultsCpt from '@/components/search/SearchResults';
+import SearchSortByMenu from '@/components/search/SearchSortByMenu';
+
+export default async function Page({ searchParams }: { searchParams: SearchParams }) {
+  let data: SearchResults | null = null;
+  let resultsType: SearchResultsType = searchParams.resultsType || 'all';
+  let versionsData: SongVersions | null = null;
+
+  if (searchParams.q) {
+    const response = await fetch(`${API_DOMAIN}/api/v2/search?q=${searchParams.q}`, {
+      cache: 'no-cache', // seconds
+    }).then((res) => res.json());
+
+    // API sometimes returns null instead of []
+    data = { ...response, Songs: !response.Songs ? [] : response.Songs };
+  }
+
+  if (searchParams.artistUuid && searchParams.songUuid) {
+    const [artistResponse, versionsRespose] = await Promise.all([
+      fetch(`${API_DOMAIN}/api/v3/artists/${searchParams.artistUuid}`, { cache: 'no-cache' }),
+      fetch(
+        `${API_DOMAIN}/api/v3/artists/${searchParams.artistUuid}/songs/${searchParams.songUuid}`,
```

Comment:
Search query parameters could look cleaner by sending artist slug rather than uuid, but the api requires artist uuid to get song versions. So we're passing artist uuid and making an additional API call to return artist name and slug, for the purpose of keeping name and slug out of `searchParams`